### PR TITLE
[Docs] Add docs of get_started, migration and overview

### PR DESCRIPTION
**Pull request by JingweiZhang12**
The migration.md is in discusse. We wait for the template draft.

### CODE REVIEW

File: docs/en/get_started.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+8. Install extra dependencies
+
+- For MOT evaluation (required):
+
+  ```shell
+  pip install git+https://github.com/JonathonLuiten/TrackEval.git
+  ```
+
+- For VOT evaluation (optional)
```
Comment (dyhBUPT) [inline docs/en/get_started.md:120]: Maybe we should remind the user that, we should install vot toolkit first, otherwise there may be an error.

File: docs/en/get_started.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+Note:
+
+a. Following the above instructions, MMTracking is installed on `dev` mode
+, any local modifications made to the code will take effect without the need to reinstall it.
+
+b. If you would like to use `opencv-python-headless` instead of `opencv-python`,
+you can install it before installing MMCV.
+
+### A from-scratch setup script
```
Comment (dyhBUPT) [inline docs/en/get_started.md:145]: Is mmclassification also needed?

File: docs/en/get_started.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+```
+python demo/{DEMO_FILE} \
+    ${CONFIG_FILE}\
+    --input ${INPUT} \
+    --checkpoint ${CHECKPOINT_FILE} \
+    [--output ${OUTPUT}] \
+    [--device ${DEVICE}] \
+    [--show] \
+    [--fps ${FPS]
```
Comment (dyhBUPT) [inline docs/en/get_started.md:196]: missing a `}`

File: docs/en/get_started.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+```
+
+
+
+## Verification
+
+To verify whether MMTracking and the required environment are installed correctly, we can run MOT, VIS, VID and SOT [demo scripts](https://github.com/open-mmlab/mmtracking/tree/master/demo/):
+
+```
+python demo/{DEMO_FILE} \
```
Comment (dyhBUPT) [inline docs/en/get_started.md:189]: missing a `$`

File: docs/en/get_started.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+```shell
+python demo/demo_mot_vis.py \
+    configs/mot/deepsort/deepsort_faster-rcnn-resnet50-fpn_8x2bs-4e_mot17halftrain_test-mot17halfval.py \
+    --input demo/demo.mp4 \
+    --output mot.mp4 \
+```
+
+**Note**:
++ Deepsort don't need to load checkpoint since the tracker don't have to be trained. However, for other MOT methods, such as byteytrack and qdtrack, you need to add an input argument `--checkpoint` in the commend script.
```
Comment (dyhBUPT) [inline docs/en/get_started.md:241]: Deepsort -> DeepSORT

Comment (dyhBUPT) [reply docs/en/get_started.md:241]: don't -> doesn't

Comment (dyhBUPT) [reply docs/en/get_started.md:241]: bytetrack -> ByteTrack
qdtrack -> QDTrack

File: docs/en/get_started.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
++ When running `demo_mot_vis.py`, we suggest you use the config containing `private`, since `private` means the MOT method doesn't need external detections.
```
Comment (dyhBUPT) [inline docs/en/get_started.md:242]: For now, all MOT methods are `private`.

File: docs/en/get_started.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+```shell
+python ./demo/demo_sot.py \
+    ./configs/sot/siamese_rpn/siamese-rpn_resnet50_8x28bs-20e_imagenetvid-imagenetdet-coco_test-lasot.py \
+    --input ${VIDEO_FILE} \
+    --checkpoint checkpoints/siamese_rpn_r50_1x_lasot_20211203_151612-da4b3c66.pth \
+    --output ${OUTPUT} \
+    --show
+```
+**Note** that the SOT demo support load the annotation of the first frame from the annotation file. You can specify an argument
```
Comment (dyhBUPT) [inline docs/en/get_started.md:265]: support load -> supports loading

File: docs/en/get_started.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+```shell
+python ./demo/demo_sot.py \
+    ./configs/sot/siamese_rpn/siamese-rpn_resnet50_8x28bs-20e_imagenetvid-imagenetdet-coco_test-lasot.py \
+    --input ${VIDEO_FILE} \
+    --checkpoint checkpoints/siamese_rpn_r50_1x_lasot_20211203_151612-da4b3c66.pth \
+    --output ${OUTPUT} \
+    --show
+```
+**Note** that the SOT demo support load the annotation of the first frame from the annotation file. You can specify an argument 
+`--gt_bbox_file` which menas the gt_bbox file path of the video. The first line annotatioin in the file will be used to initilize the tracker. If not specified, you would draw init bbox of the video manually.**
```
Comment (dyhBUPT) [inline docs/en/get_started.md:266]: menas -> means

Comment (dyhBUPT) [reply docs/en/get_started.md:266]: annotatioin -> annotation

Comment (dyhBUPT) [reply docs/en/get_started.md:266]: redundant "**"

File: docs/en/overview.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+- Video object detection (VID)
+- Single object tracking (SOT)
+- Multiple object tracking (MOT)
+- Video instance segmentation (VIS)
+
+## Major features
+
+- **The First Unified Video Perception Platform**
+
+  We are the first open source toolbox that unifies versatile video perception tasks include video object detection, multiple object tracking, single object tracking and video instance segmentation.
```
Comment (dyhBUPT) [inline docs/en/overview.md:16]: include -> including

File: docs/en/overview.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+Results and models are available in the [model zoo](docs/en/model_zoo.md).
+
+## Contributing
+
+We appreciate all contributions to improve MMTracking. Please refer to [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) for the contributing guideline and [this discussion](https://github.com/open-mmlab/mmtracking/issues/73) for development roadmap.
+
+## FAQ
+
+If you encounter any problems in the process of using MMTracking, you can firstly refer to this [faq](docs/en/notes/faq.md). If not solved, you can propose an [issue](https://github.com/open-mmlab/mmtracking/issues/)
```
Comment (dyhBUPT) [inline docs/en/overview.md:73]: faq -> FAQ

File: docs/en/overview.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+  **Simple**: MMTracking interacts with other OpenMMLab projects. It is built upon [MMDetection](https://github.com/open-mmlab/mmdetection) that we can capitalize any detector only through modifying the configs.
+
+  **Fast**: All operations run on GPUs. The training and inference speeds are faster than or comparable to other implementations.
+
+  **Strong**: We reproduce state-of-the-art models and some of them even outperform the official implementations.
+
+
+## Getting Started
+
+Please [get_started.md](docs/en/get_started.md) for the basic usage of MMTracking.
```
Comment (dyhBUPT) [inline docs/en/overview.md:33]: Please -> Please refer to

File: docs/en/overview.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
+## Introduction
+
+MMTracking is an open source video perception toolbox by PyTorch. It is a part of [OpenMMLab](https://openmmlab.com) project.
```
Comment (dyhBUPT) [inline docs/en/overview.md:3]: maybe, add the official link of PyTorch.

File: docs/en/get_started.md (hunk 2 above)
Comment (JingweiZhang12) [reply docs/en/get_started.md:145]: We add it in the requirements/runtime.txt

File: docs/en/get_started.md (hunk 1 above)
Comment (JingweiZhang12) [reply docs/en/get_started.md:120]: There are no numpy conficts  if we  perform the `pip install -r requirements/build.txt` before installing mmtrack. 
I run the install pipeline above and it's OK.

File: docs/en/migration.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+### Train
+
+### Test
+
+## Data
+
+### data structure
+
+The 1.xx versions of MMtracking add two new data structure: [TrackDataSample](mmtrack/structures/track_data_sample.py) and [ReIDDataSample](mmtrack/structures/reid_data_sample.py). These data structures wrap the annotations and predictions from one image (sequence) and used as interfaces between different components.
```
Comment (dyhBUPT) [inline docs/en/migration.md:133]: and used as -> and are used as

File: docs/en/migration.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+### dataset class
+
+The 1.xx versions of MMTracking add two base dataset class which inheient from the `BaseDataset` in MMEngine: `BaseSOTDataset` and `BaseVideoDataset`. The former is only used in SOT and the latter is used for all other tasks.
```
Comment (dyhBUPT) [inline docs/en/migration.md:137]: class -> classes

File: docs/en/migration.md [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+
+### data pipeline
+
+1. Most of the transforms on the image sequences in the 0.xx versions of MMTracking are refactored. In the 1.xx versions of MMTracking, for the image sequences, we use `TransformBroadcaster` to wrap the transformes of single image.
```
Comment (dyhBUPT) [inline docs/en/migration.md:141]: transformes -> transforms

File: docs/en/migration.md [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+
+Some tranformers on the images sequences are reserved, such as `SeqCropLikeStark`, since `TransformBroadcaster` doesn't support setting different arguments respectively for each image in the sequence.
```
Comment (dyhBUPT) [inline docs/en/migration.md:143]: transformers -> transforms

File: docs/en/migration.md [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+
+The 1.xx versions of MMtracking add `DATA_SAMPLERS` registry. You can customize different dataset samplers in the configs. Details about the samplers can be seen [here](mmtrack/datasets/samplers)
+
+## Evaluation
+
+The old versions of MMTarcking implement evaluation in the dataset class. In the 1.xx versions of MMTracking, we add `METRICS` registry. All evaluation are implemented in the metric classes registered in `METRICS`. Details can be seen [here](mmtrack/evaluation/metrics).
+
+## Visualization
+
+The 1.xx versions of MMTracking add `TrackLocalVisualizer` and `DetLocalVisualizer` which are registered in `VISUALIZER`. Compared with the 0.xx versions of MMTracking, we support the visualization of images and feature maps. Details can be seen [here](mmtrack/visualization/local_visualizer.py)
```
Comment (dyhBUPT) [inline docs/en/migration.md:165]: add a `.` at the end

File: docs/en/migration.md [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+
+Some transforms on image sequences, such as `SeqCropLikeStark`, are reserved since `TransformBroadcaster` doesn't support setting different arguments respectively for each image in the sequence.
+
+2. We pack the `VideoCollect`, `ConcatSameTypeFrames` and `SeqDefaultFormatBundle` in the old MMTracking into `PackTrackInputs` in the latest MMTracking.
+
+3. The normalizaion in the pipeline in the old MMTracking is removed and this operation is implemented in the model forward.
+
+### data sampler
+
+The 1.xx versions of MMtracking add `DATA_SAMPLERS` registry. You can customize different dataset samplers in the configs. Details about the samplers can be seen [here](mmtrack/datasets/samplers)
```
Comment (dyhBUPT) [inline docs/en/migration.md:157]: add a `.` at the end.

File: docs/en/migration.md [hunk 19]
```diff
@@ -0,0 +1,169 @@
+# Migration from MMTracking 0.xx
+
+Compared with the 0.xx versions of MMTracking, the latest 1.xx version of MMTracking has the following importrant modifications.
+
+## Overall Structures
+
+The `core` in the old versions of MMTracking is splited into `engine`, `evaluation`, `structures`, `visualization` and `model/task_moduls` in the 1.xx version of MMTracking. Details can be seen in the [user guides](docs/en/user_guides)
```
Comment (dyhBUPT) [inline docs/en/migration.md:7]: add a `.` at the end.

File: docs/en/get_started.md [hunk 20]
```diff
@@ -0,0 +1,270 @@
+## Prerequisites
+
+- Linux | macOS | Windows
+- Python 3.6+
+- PyTorch 1.3+
```
Comment (GT9505) [inline docs/en/get_started.md:5]:
```suggestion
- PyTorch 1.6+
```

File: docs/en/get_started.md [hunk 21]
```diff
@@ -0,0 +1,270 @@
...
+| :----------------: | :------------------------: | :-----------------: |
+|       master       | mmcv-full>=1.3.17, \<1.6.0 | MMDetection>=2.19.1 |
+|       0.13.0       | mmcv-full>=1.3.17, \<1.6.0 | MMDetection>=2.19.1 |
+|       0.12.0       | mmcv-full>=1.3.17, \<1.5.0 | MMDetection>=2.19.1 |
+|       0.11.0       | mmcv-full>=1.3.17, \<1.5.0 | MMDetection>=2.19.1 |
+|       0.10.0       | mmcv-full>=1.3.17, \<1.5.0 | MMDetection>=2.19.1 |
+|       0.9.0        | mmcv-full>=1.3.17, \<1.5.0 | MMDetection>=2.19.1 |
+|       0.8.0        | mmcv-full>=1.3.8, \<1.4.0  | MMDetection>=2.14.0 |
+|       0.7.0        | mmcv-full>=1.3.8, \<1.4.0  | MMDetection>=2.14.0 |
+|       0.6.0        | mmcv-full>=1.3.8, \<1.4.0  | MMDetection>=2.14.0 |
```
Comment (GT9505) [inline docs/en/get_started.md:24]: delete these 0.x version

File: docs/en/get_started.md [hunk 22]
```diff
@@ -0,0 +1,270 @@
...
+- GCC 5+
+- [MMCV](https://mmcv.readthedocs.io/en/latest/get_started/installation.html)
+- [MMEngine](https://mmengine.readthedocs.io/en/latest/get_started/installation.html)
+- [MMDetection](https://mmdetection.readthedocs.io/en/latest/get_started.html#installation)
+
+The compatible MMTracking, MMCV, and MMDetection versions are as below. Please install the correct version to avoid installation issues.
+
+| MMTracking version |        MMCV version        | MMDetection version |
+| :----------------: | :------------------------: | :-----------------: |
+|       master       | mmcv-full>=1.3.17, \<1.6.0 | MMDetection>=2.19.1 |
```
Comment (GT9505) [inline docs/en/get_started.md:16]:
```suggestion
|       master       | mmcv-full>=2.0.0, \<3.0.0 | MMDetection>=3.0.0 |
```

File: docs/en/get_started.md [hunk 23]
```diff
@@ -0,0 +1,270 @@
...
+
+- `CHECKPOINT_FILE`: The checkpoint is optional in case that you already set up the pretrained models in the config by the key `pretrains`.
+- `OUTPUT`: Output of the visualized demo. If not specified, the `--show` is obligate to show the video on the fly.
+- `DEVICE`: The device for inference. Options are `cpu` or `cuda:0`, etc.
+- `--show`: Whether show the video on the fly.
+- `FPS`: The fps of the video. Only used for show or saving the video.
+
+For VID/MOT/VIS tasks, you can also specify the argument `--score-thr` in the command script to set the threshold of score to filter bboxes.
+
+Assume that you have already downloaded the checkpoints to the directory `checkpoints/`. We provide some examples for different task:
```
[the remark this answers is not in the document]
Comment (JingweiZhang12) [reply docs/en/get_started.md:221]: OK.  I add a link to inference.md if uses wan to know more about details.

**Review by GT9505 — APPROVED** [review]

File: docs/en/overview.md [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+- [datasets](docs/en/advanced_guides/4_datasets.md)
+- [transforms](docs/en/advanced_guides/5_transforms.md)
+- [evaluation](docs/en/advanced_guides/6_evaluation.md)
+- [engine](docs/en/advanced_guides/7_engine.md)
+- [convention](docs/en/advanced_guides/8_convention.md)
+- [add modules](docs/en/advanced_guides/9_add_modules.md)
+- [add datasets](docs/en/advanced_guides/10_add_datasets.md)
+- [add transforms](docs/en/advanced_guides/11_add_transforms.md)
+- [add metrics](docs/en/advanced_guides/12_add_metrics.md)
+- [customized runtime](docs/en/advanced_guides/13_custime_runtime.md)
```
Comment (GT9505) [inline docs/en/overview.md:61]: wrong link

File: docs/en/overview.md [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+
+Results and models are available in the [model zoo](docs/en/model_zoo.md).
+
+## Contributing
+
+We appreciate all contributions to improve MMTracking. Please refer to [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) for the contributing guideline and [this discussion](https://github.com/open-mmlab/mmtracking/issues/73) for development roadmap.
+
+## FAQ
+
+If you encounter any problems in the process of using MMTracking, you can firstly refer to [FAQ](docs/en/notes/faq.md). If not solved, you can post an [issue](https://github.com/open-mmlab/mmtracking/issues/) and we will give a response as soon as possible.
```
Comment (GT9505) [inline docs/en/overview.md:73]: wrong link

File: docs/en/get_started.md [hunk 26]
```diff
@@ -0,0 +1,191 @@
...
+
+To use the default MMTracking installed in the environment rather than that you are working with, you can remove the following line in those scripts
+
+```shell
+PYTHONPATH="$(dirname $0)/..":$PYTHONPATH
+```
+
+## Verification
+
+To verify whether MMTracking and the required environment are installed correctly, we can run **one of** MOT, VIS, VID and SOT [demo scripts](https://github.com/open-mmlab/mmtracking/tree/master/demo/):
```
Comment (GT9505) [inline docs/en/get_started.md:180]: wrong link

**Review by GT9505 — APPROVED** [review]
